### DISCUSSION TITLE
sweetalert2 in livewire

### DISCUSSION THREAD

**Discussion by Shadecute** [livewire/livewire#3618] [1. Help]
i'm try to use sweetalert2 in livewire - Instead of deleting one post, all posts will be deleted
Please help me, where is my problem?

**post.list.blade**
`              <button wire:click="deleteConfirm" type="button">
                        delete
                    </button>`

**component**
`public function deleteConfirm()
    {
        $this->emit('swal', 'are u sure?', 'warning');
    }
`

and 

`    public function delete()
    {
        $this->post->delete();
    }`

my js:

`
const Swal = Swal.mixin({
    position: 'center',
    showConfirmButton: true,
})
document.addEventListener('livewire:load', () => {
    Livewire.on('swal', (message,type) => {
        Swal.fire({
            icon: type,
            text: message,
            showCancelButton: true,
        }).then((result) => {
            if (result.isConfirmed) {
              livewire.emit('delete')
            }
        })
    })
})
`

**Comment by kevinwilliam7** [comment]
I prefer to use the code below

**component**
`
public function deleteConfirm($id){
    $this->dispatchBrowserEvent('swal:deleteConfirm', [
        'type'=>'warning',
        'title'=>'are u sure?',
    ]);
}
`
and
`
public function delete($id){
    Post::where('id',$id)->delete();
}
`

then add this javascript in your post.list.blade

`
window.addEventListener('swal:deleteConfirm',event=>{
    swal({
        title: event.detail.title,
        icon: event.detail.type,
        buttons: true,
        dangerMode: true,
    })
    .then((willDelete)=>{
        if(willDelete){
            window.livewire.emit('delete', event.detail.id);
            swal({
                icon:'success',
            })
        }
    });
});
`

**Comment by mikeroq** [comment]
As an alternative I use [jantinnerezo/livewire-alert](https://github.com/jantinnerezo/livewire-alert) for sweetalert2 in Livewire. Makes it real easy.
